### PR TITLE
Taboola Bid Adapter: fix ortb2 user override issue

### PR DESCRIPTION
**Pull request by ahmadlob**
## Type of change

- [x] Bugfix
- [ ] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

user object entities in first party data (ortb2) not passed within the server generated request.
In this PR, fixed the override issue and passing what the ortb convertor generated "user" field as it is

### CODE REVIEW

**Review by dmitry-taboola — APPROVED** [review]

**Review by dmitry-taboola — APPROVED** [review]

File: test/spec/modules/taboolaBidAdapter_spec.js [hunk 1]
```diff
@@ -363,12 +363,16 @@ describe('Taboola Adapter', function () {
...
+            user: {
+              id: 'externalUserIdPassed'
+            }
           }
         }
         const res = spec.buildRequests([defaultBidRequest], bidderRequest);
         expect(res.data.bcat).to.deep.equal(bidderRequest.ortb2.bcat)
         expect(res.data.badv).to.deep.equal(bidderRequest.ortb2.badv)
         expect(res.data.wlang).to.deep.equal(bidderRequest.ortb2.wlang)
+        expect(res.data.user.id).to.deep.equal(bidderRequest.ortb2.user.id)
```
Comment (spotxslagle) [inline test/spec/modules/taboolaBidAdapter_spec.js:375]: I would also like to see a test that regs.coppa is being set.

**Review by spotxslagle — CHANGES_REQUESTED** [review]
I'd just like one small test added. Other than that, it LGTM

**Comment by ahmadlob** [comment]
@spotxslagle  added more tests, thanks!

**Review by spotxslagle — APPROVED** [review]

**Comment by ahmadlob** [comment]
@ChrisHuie When approximately this PR will be merged? thanks

**Pull request merged**